newsroom/template_filters.py

```diff
@@ -61,6 +70,84 @@ def parse_date(datetime):
     return datetime
 
 
+def get_schedule_type(start: datetime, end: datetime, all_day: bool, no_end_time: bool) -> ScheduleType:
+    duration = int((end - start).total_seconds() / 60)  # Convert seconds to minutes
+    DAY_IN_MINUTES = 24 * 60 - 1
+
+    if all_day:
+        return ScheduleType.ALL_DAY if duration == 0 else ScheduleType.MULTI_DAY
+
+    if no_end_time:
+        return ScheduleType.NO_DURATION
+
+    if duration > DAY_IN_MINUTES or start.date() != end.date():
+        return ScheduleType.MULTI_DAY
+
+    if duration == DAY_IN_MINUTES and start.date() == end.date():
+        return ScheduleType.ALL_DAY
+
+    if duration == 0 and start.time() == end.time():
+        return ScheduleType.NO_DURATION
+
+    return ScheduleType.REGULAR
+
+
+def is_item_tbc(item: dict) -> bool:
+    event_tbc = item.get("event", {}).get("_time_to_be_confirmed", False)
+    if not event_tbc:
+        planning = item.get("planning_items", [])
+        return planning[0].get("_time_to_be_confirmed", False) if planning else False
+    return event_tbc
+
+
+def format_event_datetime(item: dict) -> Optional[str]:
```

Comment:
As this is a jinja formatter, we should probably return an empty string instead of `None`